index.d.ts

```diff
@@ -2914,31 +2914,45 @@ declare namespace Shopify {
     | 'refund'
     | 'sale'
     | 'void';
-  type TransactionSourceName = 'android' | 'iphone' | 'pos' | 'web';
   type TransactionStatus = 'error' | 'failure' | 'pending' | 'success';
 
+  interface IExtendedAuthorizationAttributes {
+    standard_authorization_expires_at: string;
+    extended_authorization_expires_at: string;
+  }
+
+  interface IPaymentsRefund {
+    status: TransactionStatus;
+    acquirer_reference_number: string;
+  }
+
   interface ITransaction {
     amount: string;
-    authorization: string;
+    authorization: string | null;
+    authorization_expires_at?: string | null;
     created_at: string;
     currency: string;
-    currency_exchange_adjustment: ICurrencyExchangeAdjustment;
-    device_id: string;
-    error_code: TransactionErrorCode;
+    currency_exchange_adjustment?: ICurrencyExchangeAdjustment | null;
+    device_id: number | null;
+    error_code: TransactionErrorCode | null;
+    extended_authorization_attributes?:
+      | IExtendedAuthorizationAttributes
+      | Record<string, never>;
     gateway: string;
     id: number;
     kind: TransactionKind;
-    location_id: number;
-    message: string;
+    location_id: number | null;
```

Comment:
`null` is ok but I'm not sure about `number`. In a documentation example the value is `{ "id": 49202758 }`.

Reply:
Yeah, this one I was unsure about. That nested object in the examples was kind of suspicious, and I didn’t have any live data with `location_id` set to verify the correct type. I decided to err on the side of lesser change – I just made the field nullable, which is obviously a correct change, and left unchanged what it is in case it isn’t null. I can change this to `{ id: number } | null` if that sounds better?

For what it’s worth, the official client types this as `number | null`: https://github.com/Shopify/shopify-node-api/blob/main/src/rest-resources/2022-04/transaction.ts#L108